### PR TITLE
Pin ophyd_async to fix motor field types

### LINKED ISSUE #489: Fix issue with incorrect motor types
Currently if you run the beamline system tests you will get an error like:
```
ophyd_async.core.utils.NotConnected: 
coros: NotConnected:
    gap_motor: NotConnected:
        motor_done_move: TypeError: SR04I-MO-SERVC-01:BLGAPMTR.DMOV has type int not float
        low_limit_travel: TypeError: SR04I-MO-SERVC-01:BLGAPMTR.LLM has type float not int
        high_limit_travel: TypeError: SR04I-MO-SERVC-01:BLGAPMTR.HLM has type float not int
```
This is due to https://github.com/bluesky/ophyd-async/pull/273. We should pin `dodal` to this `ophyd_async` change to avoid this breaking in prod

## Acceptance Criteria
* The system tests pass again

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #489

### Instructions to reviewer on how to test:
1. Run the beamline system tests on main (remembering to comment out the devices from https://github.com/DiamondLightSource/dodal/issues/261) confirm you get the error in issue
2. Reinstall dodal dependencies from this branch and run again, confirm you don't get the error

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ classifiers = [
 description = "Ophyd devices and other utils that could be used across DLS beamlines"
 dependencies = [
     "ophyd",
-    "ophyd-async>=0.3a2",
+    "ophyd-async @ git+https://github.com/bluesky/ophyd-async.git@c17a372d9d3389bad030673f333e04f1e8659b0c", # Needed for https://github.com/bluesky/ophyd-async/pull/273, unpin when merged
```
Comment (Tom-Willemsen) [inline pyproject.toml:16]: I guess you mean "when included in a release" as opposed to "when merged", since it's already merged?

Comment (DominicOram) [reply pyproject.toml:16]: Could we just do a new alpha release today @callumforrester or @abbiemery?

Comment (DominicOram) [reply pyproject.toml:16]: We shouldn't do a release of `ophyd_async` until https://github.com/DiamondLightSource/dodal/issues/492 is sorted

Comment (Tom-Willemsen) [reply pyproject.toml:16]: Oh god we're getting into a horrible web of everything depending on everything else again...

As a side note, the `i03` system tests don't fail for me on either `main` or this branch :shrug: . But it looks like it should be a valid fix...

Comment (DominicOram) [reply pyproject.toml:16]: It's perfectly possible I'm just going mad with my dev environment, trying to juggle about 4 different issues at the same time today

**Comment by DiamondJoseph** [comment]
Closing: fixed with the new ophyd-async pre-release

**Pull request closed without merging**